synapse/storage/databases/main/room.py

```diff
@@ -1274,6 +1274,29 @@ async def is_partial_state_room(self, room_id: str) -> bool:
 
         return entry is not None
 
+    @cachedList(cached_method_name="is_partial_state_room", list_name="room_ids")
+    async def is_partial_state_rooms(
+        self, room_ids: StrCollection
+    ) -> Mapping[str, bool]:
```

Comment:
I wonder if we should *only* return rooms which are partial stated, in order to save room in the cache. This would mean we do something like `Mapping[str, Optional[Literal[True]]]`. I'm not 100% sure how we would handle this for the single room case though.

Reply:
I.e. we really want a query that gives a `Set[str]` for room IDs that we then keep up-to-date.

Reply:
I think this is a future optimization and we shouldn't wait for it to fix the issue.